js/sign.js

```diff
@@ -0,0 +1,93 @@
+const userid = document.querySelector("#email");
+const userpw = document.querySelector("#password");
+const userpwCheck = document.querySelector("#password-check");
+const emailError = document.querySelector(".emailError");
+const pwError = document.querySelector(".pwError");
+const signInButton = document.querySelector(".sign-in-button");
+const eye = document.querySelectorAll(".eye");
+const lineThrough = document.querySelectorAll(".line-through");
+
+function validEmail(obj) {
+  if (obj.value === "") {
+    emailError.textContent = "이메일을 입력해주세요.";
+    emailError.style.color = "red";
+    userid.style.border = "1px solid red";
+  } else if (validEmailCheck(obj) == false) {
+    console.log(obj.value);
+    emailError.textContent = "올바른 이메일 주소가 아닙니다.";
+    emailError.style.color = "red";
+    userid.style.border = "1px solid red";
+  } else {
+    emailError.textContent = "";
+    userid.style.border = "";
+  }
+}
+
+function validPassword(obj) {
+  if (obj.value === "") {
+    pwError.textContent = "비밀번호를 입력해주세요.";
+    pwError.style.color = "red";
+    userpw.style.border = "1px solid red";
+  } else {
+    pwError.textContent = "";
+    userpw.style.border = "";
+  }
+}
+
+function goUrl() {
+  if (userid.value === "test@codeit.com" && userpw.value === "codeit101") {
+    let link = "folder/";
+    location.href = link;
+  } else {
+    emailError.textContent = "이메일을 확인해주세요.";
+    emailError.style.color = "red";
+    userid.style.border = "1px solid red";
+    pwError.textContent = "비밀번호를 확인해주세요.";
+    pwError.style.color = "red";
+    userpw.style.border = "1px solid red";
+  }
+}
+
+function enterLogin(e) {
+  if (e.keyCode === 13) {
+    goUrl();
+  }
+}
+
+function validEmailCheck(obj) {
+  var pattern =
+    /^[0-9a-zA-Z]([-_.]?[0-9a-zA-Z])*@[0-9a-zA-Z]([-_.]?[0-9a-zA-Z])*.[a-zA-Z]{2,3}$/i;
+  return obj.value.match(pattern) != null;
```

Comment:
꿀팁 드리자면, 아래와 같이 !! 을 사용하면 됩니다.
```suggestion
  return !!obj.value.match(pattern);
```

!가 두 번 들어가게돼서 Boolean 값으로 변환되어요.
<img width="379" alt="image" src="https://github.com/codeit-bootcamp-frontend/4-Weekly-Mission/assets/154956732/2e27d38b-1178-4c4a-a8d8-85a73912b026">